python/ray/workflow/http_event_provider.py

```diff
@@ -0,0 +1,263 @@
+import asyncio
+from typing import Dict
+from fastapi import FastAPI, Request
+from fastapi.responses import JSONResponse
+
+import ray
+from ray import serve
+from ray.workflow import common, workflow_context, workflow_access
+from ray.workflow.event_listener import EventListener
+from ray.workflow.common import Event
+
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class WorkflowEventHandleError(Exception):
+    """Raise when event processing failed"""
+
+    def __init__(self, workflow_id: str, what_happened: str):
+        self.message = (
+            f"Workflow[id={workflow_id}] HTTP event handle failed: {what_happened}"
+        )
+        super().__init__(self.message)
+
+
+app = FastAPI()
```

Comment:
@iycheng We follow the example usage of `fastAPI()` with `Serve` from this Ray Serve [page](url)(https://docs.ray.io/en/latest/serve/end_to_end_tutorial.html#adding-functionality-with-fastapi).

It seems that we need` app` to be defined before the class HTTPEventProvider so that we can use it for decoration for the class.
```
@serve.ingress(app)
class HTTPEventProvider
```
and the decoration of the function `send_event()`:
```
@app.post("/send_event/{workflow_id}")
async def send_event(self, workflow_id: str, req: Request) -> JSONResponse:
```

It is not clear that we could move `app = fastAPI()` inside the `HTTPEventProvider `class.